### PR TITLE
[NEW] Add "Send Crash Report" option

### PR DESCRIPTION
**Pull request by djorkaeffalexandre**
User can set `Send Crash Report`.
Default value is `false`.

### CODE REVIEW

**Comment by filipedelimabrito** [comment]
Are you guys using analytics on the app too?

**Comment by djorkaeffalexandre** [comment]
> Are you guys using analytics on the app too?

yes!

File: app/views/SettingsView/index.js [hunk 1]
```diff
@@ -83,6 +98,17 @@ export default class SettingsView extends React.Component {
 		);
 	}
 
+	renderCrashlyticsSwitch = () => {
```
Comment (diegolmello) [inline app/views/SettingsView/index.js]: We should do it same way as iOS native does:

- "Send Crash Report"
- A description of it



Also, remove "Crashlytics" word from other places like methods, actions, etc.

Comment (djorkaeffalexandre) [reply app/views/SettingsView/index.js]: nice!

Comment (djorkaeffalexandre) [reply app/views/SettingsView/index.js]: fixed.

**Comment by filipedelimabrito** [comment]
@djorkaeffalexandre Thanks for replying. I am not sure if you guys are getting the analytics data only when the `Send Crash Report` is enabled from now, but I'd like to say that from both native iOS and Android apps we are doing so. It [brings some questions with GDPR](https://github.com/RocketChat/Rocket.Chat.Android/issues/1757) too - jfyi.

That said, is this PR being applied only for Crashlytics or Analytics (or both of them)?

**Comment by djorkaeffalexandre** [comment]
@filipedelimabrito only for crashlytics. 😁

**Comment by filipedelimabrito** [comment]
@djorkaeffalexandre 😬

@diegolmello we need to start getting analytics data only after the user consent.

**Comment by diegolmello** [comment]
> @diegolmello we need to start getting analytics data only after the user consent.

@filipedelimabrito I couldn't find `Send analytics` on iOS native or `Send crash report` on Android native.
I couldn't find these options on onboarding on native apps neither, as the link you sent.

I think this PR does what the title says and we could implement `Analytics collection` later: https://rnfirebase.io/docs/v5.x.x/analytics/reference/analytics#setAnalyticsCollectionEnabled

**Comment by diegolmello** [comment]
By testing, we figured Firebase's API could be better: https://rnfirebase.io/docs/v5.x.x/crashlytics/reference/crashlytics#enableCrashlyticsCollection

It forces us to start the app disabled and enable inside the code with `enableCrashlyticsCollection` method.
What if the user wants to disable crash report?
Should we ask him to close the app and reopen?
That's not a good UX.

Let's close this PR and try Bugsnag in the future.
At least their API _seems_ to be better than this: https://docs.bugsnag.com/platforms/react-native/react-native/#disabling-error-reporting

Thanks anyway, @djorkaeffalexandre 👍

**Comment by filipedelimabrito** [comment]
@diegolmello I can refer to you the code on native ones if you want to confirm it - actually the display text isn't good on iOS native (it only tells the users about the crash reports while it serves to opt in and out the analytics too), on Android you can see it (the analytics I started talking about) clearly on the settings screen (let me attach an image here).  

Sorry if it has disturbed you but I think that since this PR is implementing a collect/tracking option it worths bringing this **initial** discussion here - to you and @djorkaeffalexandre. 


Screen Shot 2019-07-23 at 18 29 29

**Comment by diegolmello** [comment]
@filipedelimabrito You're right.
I just followed what the iOS label says.
Anyway, the lib API is not good and we're planning to use Bugsnag.

**Pull request closed without merging**